### PR TITLE
resource/nifcloud_db_instance: Add resource

### LINKED ISSUE #16: nifcloud_db_instance: Add resource
### Overview

Add a new resource `nifcloud_db_instance`.

### PR DESCRIPTION
**Pull request by fuku2014**
## Description

- Add nifcloud_db_instance resource.
- Closes #16 

## How Has This Been Tested?

- [x]  Buld with `make install` command, and Apply example.

```:console
make install
cd examples/db_instance
terraform init
terraform plan
terraform apply
```

## Types of Changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (improves code quality but has no user-facing effect)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

### CODE REVIEW

File: nifcloud/acc/db_instance_test.go [hunk 1]
```diff
@@ -0,0 +1,483 @@
...
+		if nifcloud.StringValue(foundDBInstance.DBInstanceIdentifier) != saved.Primary.ID {
+			return fmt.Errorf("db instance does not found in cloud: %s", saved.Primary.ID)
+		}
+
+		*dbInstance = foundDBInstance
+		return nil
+	}
+}
+
+func testAccCheckDBInstanceReplicaExists(n string) resource.TestCheckFunc {
```
Comment (aokumasan) [inline nifcloud/acc/db_instance_test.go:191]: This function's logic is duplicated with`testAccCheckDBInstanceRestoreExists`...?

Comment (fuku2014) [reply nifcloud/acc/db_instance_test.go]: @aokumasan 

Fixed it!

Comment (aokumasan) [reply nifcloud/acc/db_instance_test.go]: Thanks!!

**Comment by aokumasan** [comment]
LGTM 👍

**Pull request merged**